src/components/BurgerIngredientCard/BurgerIngredientCard.jsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { CurrencyIcon } from '@ya.praktikum/react-developer-burger-ui-components';
+import styles from './BurgerIngredientCard.module.css';
+import PropTypes from 'prop-types';
+import Modal from '../Modal/Modal';
+
+const BurgerIngredientCard = (props) => {
+    const [isModalOpen, openModal] = React.useState(false);
+
+    return (
+        <>
+            <div
+                onClick={() => openModal(true)}
+                key={props._id}
+                type={props.type}
+                className={`${styles.wrapper} 
+            pt-6`} >
+                <div>
+                    <img src={props.image} />
+                </div>
+                <div className={styles.inner}>
+                    <span>{props.fat}</span>
+                    <CurrencyIcon type="primary" />
+                </div>
+                <p>{props.name}</p>
+            </div>
+            {
+                isModalOpen && <Modal
+                    onClick={() => openModal(false)}
+                    isModalOpen={isModalOpen}
+                    title={'Детали ингридиента'}>
+                    123
+                </Modal>
+            }
```

Comment:
```suggestion
            {
                isModalOpen && <Modal
                    setIsModalOpen={() => setIsModalOpen(false)}
                    title={'Детали ингридиента'}>
                    123
                </Modal>
            }
```
Пробрасываем сеттер и убираем ненужный пропс isModalOpen.